pilot/pkg/bootstrap/server.go

```diff
@@ -1191,3 +1204,56 @@ func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
 
 	return true
 }
+
+func (s *Server) initCertController(args *PilotArgs) error {
+	var err error
+	var svcNames []string
+	var svcNamespaces []string
+	if !s.mesh.K8SCertificateSetting.Enabled {
```

Comment:
How can we make sure `s.mesh.K8SCertificateSetting` is not nil?

Reply:
I think we donot need `Enabled`

Reply:
- Added a check for nil s.mesh.K8SCertificateSetting.
- Enabled is to explicitly enable/disable k8s certificate generation.

Reply:
For 2, personally, i think the pointer type is enough

Reply:
The "Enabled" field provides a better readability.